### ISSUE TITLE
Allow [modify_side] color=hex,hex,hex,hex in [color_range] format (GNA #23815)

### ISSUE THREAD

**Issue by wesnoth-bugs** [wesnoth/wesnoth#1501]
**[Original submission](https://gna.org/bugs/index.php?23815)** by ravana on 2015-08-22

Currently custom team colors require defining [color_range] in order to have id to use in [modify_side].

However [color_range] is toplevel tag and its content is not sent in multiplayer so currently people who do not have color with specified id see red teamcolor and green village flag for those sides.

Having color defined by hex values directly would remove the need for predefined [color_range] tags and allow better control over custom team colors.

Additionally, replacing side color should also replace the village flag color of that side.

(Reproduced on any)
Release: 1.13
Priority: 5 - Normal
Severity: 3 - Normal

**Comment by ProditorMagnus** [comment]
It is not support with this key/value combination at least. Havent investigated its cpp source.

**Comment by CelticMinstrel** [comment]
Oh, in `[modify_side]`, huh. I wonder if it works directly in `[side]`...

**Comment by gfgtdf** [comment]
what works in [modify_side] shodul also work in [side] and the other way around, if that's not the case i'd call it a bug.


And just to make sure that i understood you correctly, in your suggestion all 'hex' are rgb string so the suggested format is for example `color=D1620D,FFFFFF,000000,FF0000`?

**Comment by ProditorMagnus** [comment]
Yes, this is same format that color_range uses. Not exactly sure how each of these colors are used.

**Comment by gfgtdf** [comment]
this example is is actually directly copied from the mainline rolor_range file.

**Comment by CelticMinstrel** [comment]
Hmm. Maybe it's average, max, min, text? Though that doesn't seem to fit your example either...
